iotdb-core/datanode/src/main/java/org/apache/iotdb/db/pipe/event/common/tablet/TabletInsertionDataContainer.java

```diff
@@ -105,7 +115,14 @@ private void parse(InsertRowNode insertRowNode, String pattern) {
 
     this.deviceId = insertRowNode.getDevicePath().getFullPath();
     this.isAligned = insertRowNode.isAligned();
-    this.timestampColumn = new long[] {insertRowNode.getTime()};
+
+    final long[] originTimestampColumn = new long[] {insertRowNode.getTime()};
+    List<Integer> rowIndexList =
```

Comment:
Maybe this procedure can be accelerated because of the originTimestampColumn is sorted?